sunbird_service_api_test/src/test/java/org/sunbird/integration/test/course/batch/CreateCourseBatchTest.java

```diff
@@ -0,0 +1,179 @@
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_ENROLLMENTTYPE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_ENROLLMENTTYPE,
+        false,
+        HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITHOUT_START_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_PAST_START_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_COURSE_ID, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_PAST_END_DATE, false, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_END_DATE_BEFORE_START_DATE,
+        false,
+        HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_CREATED_FOR, true, HttpStatus.BAD_REQUEST
+      },
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_FAILURE_WITH_INVALID_MENTOR, true, HttpStatus.BAD_REQUEST
+      }
+    };
+  }
+
+  @Test(dataProvider = "createCourseBatchFailureDataProvider")
+  @CitrusParameters({"testName", "isCourseIdRequired", "httpStatusCode"})
+  @CitrusTest
+  public void testCreateCourseBatchFailure(
+      String testName, boolean isCourseIdRequired, HttpStatus httpStatusCode) {
+    getTestCase().setName(testName);
+    beforeTest(isCourseIdRequired);
+    getAuthToken(this, true);
+    variable("startDate", TODAY_DATE);
+    performPostTest(
+        this,
+        TEMPLATE_DIR,
+        testName,
+        getCreateCourseBatchUrl(),
+        REQUEST_JSON,
+        MediaType.APPLICATION_JSON,
+        true,
+        httpStatusCode,
+        RESPONSE_JSON);
+  }
+
+  @DataProvider(name = "createCourseBatchSuccessDataProvider")
+  public Object[][] createCourseBatchSuccessDataProvider() {
+    return new Object[][] {
+      new Object[] {TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY, HttpStatus.OK},
+      new Object[] {TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_OPEN, HttpStatus.OK},
+      new Object[] {
+        TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY_WITH_CREATED_FOR, HttpStatus.OK
+      },
+      new Object[] {TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY_WITH_MENTORS, HttpStatus.OK}
+    };
+  }
+
+  @Test(dataProvider = "createCourseBatchSuccessDataProvider")
+  @CitrusParameters({"testName", "httpStatusCode"})
+  @CitrusTest
+  public void testCreateCourseBatchSuccess(String testName, HttpStatus httpStatusCode) {
+    getTestCase().setName(testName);
+    getAuthToken(this, true);
+    beforeTest(true);
+    variable("startDate", TODAY_DATE);
+
+    if (testName.equalsIgnoreCase(
+        TEST_NAME_CREATE_COURSE_BATCH_SUCCESS_INVITE_ONLY_WITH_CREATED_FOR)) createOrg();
```

Comment:
In data provider add arguments for canCreateOrg, canCreateUser and use them in beforeTest to create necessary resources. Avoid if checks on test name.